src/Vault.sol

```diff
@@ -783,11 +809,45 @@ contract Vault {
         }
     }
 
+    /*///////////////////////////////////////////////////////////////
+                    ASSET MANAGEMENT LOGIC
+    ///////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Calls external action handlers to execute and interact with external logic.
+     * @dev Similar to flash loans, this function optimistically calls external logic and checks for the vault state at the very end.
+     * @param actionHandler the address of the action handler to call
+     * @param actionData a bytes object containing two actionAssetData structs, an address array and a bytes array
+     */
+    function vaultManagementAction(address actionHandler, bytes calldata actionData) public onlyOwner {
```

Comment:
Where are the tests?
- Success
- Revert: "VL_VMA: Action is not allowlisted"
- Revert: "VMA: coll. value too low"

Reply:
in 16_ActionMultiCall.t.sol

Reply:
Plz ser 😮‍💨